### ISSUE TITLE
NormalizationTransformer.untransform edge case for batch_size = 1

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1821]
Thanks to a bug-report on gitter, we have this interesting failing test case:

```
  def test_normalization_batch_one(self):                              
    """This test is based on a user bug report.
          
    When `batch_size==1` and `n_tasks > 1`,
    `NormalizationTransformer.untransform` had a weird edge case where 
    it would return a larger output that input.
    """ 
    n_samples = 11                                                     
    n_features = 5 
    n_tasks = 2                                                        
    train_dataset = dc.data.NumpyDataset(np.random.rand(n_samples, n_features), np.random.rand(n_samples, n_tasks))                            
    
    transformer = dc.trans.NormalizationTransformer(                   
            transform_y=True, dataset=train_dataset, move_mean=True)   
    
    # batch of size 1 shape                                            
    batch = np.random.rand(1, 2, 1)                                    
    output = transformer.untransform(batch)                            
      
    assert len(batch) == len(output)
```

This currently fails because of this batch of code in `NormalizationTransformer.untransform` (https://github.com/deepchem/deepchem/blob/master/deepchem/trans/transformers.py#L200):

```
for dim in z_shape:
  if dim != n_tasks and dim == 1:
    y_min = np.expand_dims(y_min, -1)
    y_max = np.expand_dims(y_max, -1)
```

The edge case occurs when `batch_size==1` and it erroneously expands the batch dimension outwards. This can be fixed by preventing expansion from happening on the batch dimension with an explicit guard:

```
# Get the reversed shape of z: (..., n_tasks, batch_size)        
z_shape.reverse()
# Find the task dimension of z
for ind, dim in enumerate(z_shape):
  if ind < (len(z_shape) - 1) and dim == 1:
    # Prevent broadcasting on wrong dimension                    
    y_stds = np.expand_dims(y_stds, -1)
    y_means = np.expand_dims(y_means, -1)
```

I've made this fix in #1792 and will upstream it in a later PR

**Comment by mcloughlin2** [comment]
I just ran into this bug while running predictions off a GraphConvModel in DeepChem 2.6.1. It looks like Bharath's fix never made it into a production release.
